### PR TITLE
Testing part update

### PR DESCRIPTION
**Pull request by JihongJeong**
I added some testing part to check the problem of test score. You can also see how the model predicted the data using print function at line:82, the raw_logits mean the mlp head result before the classification.

### CODE REVIEW

**Comment by justin4ai** [comment]
Wait, ```detector-tmp``` is a branch for v.01 as far as I remember but the batch time for training seems to be the one for v.02 (17 seconds in RTX 3060ti). **I'd like to double check** ```detector-tmp``` branch is using the old(v.01) architecture since you want me to merge it into ```detector```, which is a stable version for v.01.

**Comment by JihongJeong** [comment]
Yes, this one is for v.01, it doesn't use the v.02 architecture. It maybe because of the test session. For me, it takes 3 sec for each 10 iterations.

**Comment by justin4ai** [comment]
Alright I get that, still have no idea about the reason it takes 17 seconds for 10 iterations which is significantly different from yours even given your machine is better.

image

By the way, seemingly we can say testing code has no problem. Probably it's because fake images in the given test set are so realistic that our model cannot get it as we expected. We - especially you lol - must come up with the solution.

This still doesn't explain why even training score sucks in your machine. I will investigate it through training on the same dataset to yours. 

Just please check ```detector-tmp``` is on v.01 architecture once more and leave a comment. Then I will immediately merge it.

**Comment by justin4ai** [comment]
Also I sent a request for the access to your shared folder an hour ago. Would you mind accepting it and leave the link here again?

**Comment by JihongJeong** [comment]
I again committed the testing part. This time it will show inference time of both train part and test part. Can you check the inference time again and compare with the previous one(from the 'detector' branch)?
Also, I access you to share the google drive

**Comment by justin4ai** [comment]
image

Now I see. Testing process is quite heavy. Since there is a huge gap of the number of iterations for each epoch between us, I cannot make it into like ```if train_idx%300 = 0 : print train acc``` such that it prints the test log out every 300th iteration (for example).

Please come up with a good idea for this that can **flexibly, adaptively and periodically** print the test log out depending on the size of train set or the number of total iterations. If this is done, then let's merge it into ```detector```.

And also please **leave the link** for the shared folder in this conversation once again.

**Comment by justin4ai** [comment]
Besides all of those changes in log printing stuff need to be applied to ```FID-Transformer.v02``` as well, you know. A lot of work, but hopefully it will pay dividends in the long run I believe brother! Appreciate your dedicated work sincerely.

**Comment by JihongJeong** [comment]
https://drive.google.com/file/d/1eozcW0NdBS6p-Cv3Vg6F84ksch09DnDu/view?usp=sharing

**Comment by justin4ai** [comment]
Welp, brother would you mind uploading only dataset files believing our code files are all the same since they are hopefully well managed by Git? I always turn on my VPN for accessing to GPU server so downloading 2.3G might take a year lool. Sorry for this! 

And also after re-uploading, please right-click the folder and click ```share``` then change the permission to ```public with those who has the link ```in advance.

**Comment by JihongJeong** [comment]
https://drive.google.com/drive/folders/1kkKry6RN-8QuiWEGv1nlCetPowrc6aXO?usp=sharing

File: train.py [hunk 1]
```diff
@@ -138,6 +145,7 @@ def main(args):
     parser.add_argument('--save_path', type=str, default = "./checkpoints/", help="Path to the dataset")
     parser.add_argument('--num_epochs', type=int, default=50, help="Number of epochs to train")
     parser.add_argument('--batch_size', type=int, default=16, help="Mini batch size")
+    parser.add_argument('--test_interval', type=int, default=10, help="interval of test session during training")
```
Comment (justin4ai) [inline train.py:152]: Aight this is enough for now. Let me know if that's it so that I can immediately merge your work.

**Comment by JihongJeong** [comment]
Okay, I also upload only source code.

**Comment by justin4ai** [comment]
> https://drive.google.com/drive/folders/1kkKry6RN-8QuiWEGv1nlCetPowrc6aXO?usp=sharing

Folders under ```datasets/train``` are all empty. Are they being uploaded?

**Comment by justin4ai** [comment]
> Okay, I also upload only source code.

Nopee I mean please upload only datasets/train! Source code is not needed since we share the same repository well managed by Git. Sorry for your confusion.

File: train.py (hunk 1 above)
Comment (JihongJeong) [reply train.py:152]: Sure. Also to add, default = 10 means the testing session will run 10 times for each epochs. For you it might be 1 test session for each 109 batches

**Comment by JihongJeong** [comment]
You mean 'train/generated'? 'train/real' will take similar size with the previous one.

File: train.py (hunk 1 above)
Comment (justin4ai) [reply train.py:152]: > Sure. Also to add, default = 10 means the testing session will run 10 times for each epochs. For you it might be 1 test session for each 109 batches

Would you more elaborate on this? I thought it is literally **interval** - if I set it to N, log printing for test set is carried out every Nth iteration. But what you mean by `default = 10 means the testing session will run 10 times for each epochs.`? You mean N is equal to the total number of test session within one epoch?

**Comment by justin4ai** [comment]
> You mean 'train/generated'? 'train/real' will take similar size with the previous one.

How come 2k images in total are 2.3G sized? If it's true I have no choice for sure. Is my understand correct?

**Comment by JihongJeong** [comment]
Oh, In the code, we load 70k images and random sample 1k images to make train dataset.

**Comment by JihongJeong** [comment]
Now I recognized this. Ah, this might be critical to the training. Is this might be the reason why our trained model always predict the test image as 'generated'?

**Comment by justin4ai** [comment]
> Oh, In the code, we load 70k images and random sample 1k images to make train dataset.

Oh lord then I won't be able to reproduce the same environment to yours. Hmmm

And it is really weird since my dataset doesn't get sampled for 1k images although we use the completely same code? Any explanation about this?

**Comment by justin4ai** [comment]
> Now I recognized this. Ah, this might be critical to the training. Is this might be the reason why our trained model always predict the test image as 'generated'?

Aight I don't know what it is but hope you figured out. Get me updated anytime! I might not be able to give you back for today anymore but at least I will read all of yours. Just keep hustling brother!!

**Comment by JihongJeong** [comment]
> > Now I recognized this. Ah, this might be critical to the training. Is this might be the reason why our trained model always predict the test image as 'generated'?
> 
> Aight I don't know what it is but hope you figured out. Get me updated anytime! I might not be able to give you back for today anymore but at least I will read all of yours. Just keep hustling brother!!

I see you have 1097 iterations for train part. From this, the whole dataset size might be 1097*(10/8)*16 = 21940. Is this right? or It should be 40k

File: train.py (hunk 1 above)
Comment (JihongJeong) [reply train.py:152]: line:52  test_interval = int(len(train_dataloader)/args.test_interval)
here I set test_interval. So, if you have 1000 iteration for training and set args.test_interval to 10, testing session will appear for each 100 (=1000/10) iterations

**Comment by justin4ai** [comment]
> > > Now I recognized this. Ah, this might be critical to the training. Is this might be the reason why our trained model always predict the test image as 'generated'?
> > 
> > 
> > Aight I don't know what it is but hope you figured out. Get me updated anytime! I might not be able to give you back for today anymore but at least I will read all of yours. Just keep hustling brother!!
> 
> I see you have 1097 iterations for train part. From this, the whole dataset size might be 1097*(10/8)*16 = 21940. Is this right? or It should be 40k

Wait that's what I've been curious about. Definitely 40K

**Comment by JihongJeong** [comment]
Please train the latest commit same as before and check this real data loading part was really matter. Not right now, at you convenience

File: train.py (hunk 1 above)
Comment (justin4ai) [reply train.py:152]: Gotcha. Beautiful!

**Comment by justin4ai** [comment]
image
This error occurs! Hot fix is needed.

**Comment by JihongJeong** [comment]
> image This error occurs! Hot fix is needed.

Hotfixed the dataloading part. This occurs because the generated datasets are larger than the real datasets.

**Comment by justin4ai** [comment]
I merge this work since it is verified. Thanks!
